charts/kong/dispatch-transformer/handler.lua

```diff
@@ -198,6 +198,42 @@ local function insert_header_to_payload(conf, result)
   return result
 end
 
+local function insert_http_context_to_payload(conf, result)
+  if conf.enable.http_context then
+    local http_context = ngx.req.get_headers()
+
+    -- nginx variables
+    local fields = {
+      "args",
+      "request",
+      "request_uri",
+      "scheme",
+      "server_protocol",
+      "upstream_uri",
+      "uri",
+    }
+
+    for _, v in pairs(fields) do
+      local field = string.gsub(v, "_", "-")
+      if not http_context[field] then
+        http_context[field] = ngx.var[v]
+      end
+    end
+
+    -- Need to extract method out of request since always changed to POST
+    if not http_context["method"] then
+      local method = string.match(ngx.var.request, "%a+")
+      if method then
+        http_context["method"] = method
+      end
+    end
+
+    result[conf.substitute.http_context] = http_context
+    ngx.log(ngx.DEBUG, "after insert_http_context: payload: " .. cjson.encode(result))
```

Comment:
Could this potentially log sensitive information like cookies?

Reply:
Good catch. It could potentially log out cookies. I'll edit it out.